worktree_status.go

```diff
@@ -321,24 +321,19 @@ func (w *Worktree) AddWithOptions(opts *AddOptions) error {
 	}
 
 	if opts.All {
-		_, err := w.doAdd(".", w.Excludes)
+		_, err := w.doAdd(".", w.Excludes, false)
 		return err
 	}
 
 	if opts.Glob != "" {
 		return w.AddGlob(opts.Glob)
 	}
 
-	_, err := w.Add(opts.Path)
+	_, err := w.doAdd(opts.Path, make([]gitignore.Pattern, 0), opts.SkipStatus)
```

Comment:
@pjbgf before my change AddWithOptions for a specific file passed empty ignore pattern slice so a file meant to be ignored will be added. I preferred not to change/break this behavior although I'm not sure it's the right option. 
Do you think we should keep it as is?

Reply:
Or probably you mean file ignored by .gitignore and not excluded?

Reply:
ok, got you. We are not calling diffStagingWithWorktree which exclude the files specified in .gitingore because we are not calling Status(). 
So when skipStatus=true  if passing a path of file that meant to be ignored it will be added and if skipStatus=false it will not added. I will add a test and comment as you asked

Reply:
Thank you for following this through.